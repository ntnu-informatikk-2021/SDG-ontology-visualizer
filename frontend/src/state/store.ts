import { createStore } from 'redux';
import apiErrorReducer from './reducers/apiErrorReducer';

const store = createStore(apiErrorReducer);

export default store;
