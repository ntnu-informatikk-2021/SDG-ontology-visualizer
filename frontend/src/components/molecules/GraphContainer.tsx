import { Flex, Stack } from '@chakra-ui/react';
import React, { useState } from 'react';
import { useSelector } from 'react-redux';
import { isSubgoal, isWithinCorrelationLimit } from '../../common/node';
import { RootState } from '../../state/store';
import { D3Edge } from '../../types/d3/simulation';
import { GraphEdge, GraphNode } from '../../types/ontologyTypes';
import Graph from '../atoms/Graph';
import GraphToolBar from './GraphToolbar';
import GraphDescriptions from './GraphDescriptions';

const GraphContainer: React.FC = () => {
  const [showSubgoals, setShowSubgoals] = useState<boolean>(false);
  const [positiveConnectionChoice, setPositiveConnectionChoice] = useState<number>(0);
  const [negativeConnectionChoice, setNegativeConnectionChoice] = useState<number>(0);
  const [unlockNodes, setUnlockNodes] = useState<boolean>(false);
  const [edgeLabelsVisible, setEdgeLabelsVisible] = useState<boolean>(true);
  const { isFullscreen } = useSelector((state: RootState) => state.fullscreenStatus);

  const filterSubgoals = () => {
    setShowSubgoals(!showSubgoals);
  };

  const nodeFilter = (node: GraphNode): boolean => {
    if (!showSubgoals && isSubgoal(node)) return false;
    return true;
  };
  const edgeFilter = (edge: D3Edge | GraphEdge): boolean => {
    if (!isWithinCorrelationLimit(edge, positiveConnectionChoice, negativeConnectionChoice))
      return false;
    return true;
  };

  return (
    <Stack
      h={isFullscreen ? '100vh' : '65vh'}
      w={isFullscreen ? '100vw' : ''}
      position={isFullscreen ? 'absolute' : 'static'}
      top="0px"
      left="0px"
    >
      <GraphToolBar
        onSubgoalFilter={filterSubgoals}
        onPositiveConnectionFilter={setPositiveConnectionChoice}
        onNegativeConnectionFilter={setNegativeConnectionChoice}
        onUnlockNodes={setUnlockNodes}
        onEdgeLabelsVisible={setEdgeLabelsVisible}
      />
      <Flex mt={isFullscreen ? '0 !important' : ''} h="100%" justify="space-between">
        <Graph
          nodeFilter={nodeFilter}
          edgeFilter={edgeFilter}
          unlockAllNodes={unlockNodes}
          edgeLabelsVisible={edgeLabelsVisible}
        />
        <GraphDescriptions float={isFullscreen} />
      </Flex>
    </Stack>
  );
};

export default GraphContainer;
