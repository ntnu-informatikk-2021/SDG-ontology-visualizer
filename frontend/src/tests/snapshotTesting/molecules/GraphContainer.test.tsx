import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import GraphContainer from '../../../components/molecules/GraphContainer';
import store from '../../../state/store';

it('renders when there are no items', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <GraphContainer />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
