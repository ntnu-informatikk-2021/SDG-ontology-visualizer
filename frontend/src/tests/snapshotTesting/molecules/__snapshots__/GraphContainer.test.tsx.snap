// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders when there are no items 1`] = `
<div
  className="css-k008qs"
>
  <div
    className="css-253icu"
  >
    <svg
      height={568}
      id="svgGraph"
      width={819.2}
    />
  </div>
  <div
    className="chakra-stack css-142b4vo"
  >
    <h3
      className="chakra-heading css-0"
    >
      Filter
    </h3>
    <label
      className="chakra-checkbox css-1uiwwan"
    >
      <input
        checked={false}
        className="chakra-checkbox__input"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        style={
          Object {
            "border": "0px",
            "clip": "rect(0px, 0px, 0px, 0px)",
            "height": "1px",
            "margin": "-1px",
            "overflow": "hidden",
            "padding": "0px",
            "position": "absolute",
            "whiteSpace": "nowrap",
            "width": "1px",
          }
        }
        type="checkbox"
      />
      <span
        aria-hidden={true}
        className="chakra-checkbox__control css-dnty2r"
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
      />
      <span
        className="chakra-checkbox__label css-1oeb2oe"
        onMouseDown={[Function]}
        onTouchStart={[Function]}
      >
        Vis delm√•l
      </span>
    </label>
  </div>
</div>
`;
